### ISSUE TITLE
Jupyter function registration throws due to type annotation serialization

### ISSUE THREAD

**Issue by andyvandy** [cunnane/xloil#71]
I'm encountering errors when running the jupyter function registration example from the [docs ](https://xloil.readthedocs.io/en/latest/xlOil_Python/Jupyter.html ) when using python 3.11, 32bit with xloil 0.17.13. 
The error can be reproduced with test notebook tests/ManualSheets/python/TestJupyterConnection.ipynb 

```
@xloil.func
def jptest(x):
    return f"Jupyter says {x}"
```

I can get the registration to work for functions with 0 args so I believe the root issues are tied to Arg object de/serialization.

On the jupyter side, registration fails when arguments ( or the fn return ) are annotated with str type.  The recursive serialization fails since  `str.__new__.__dict__` doesn't exist

I think special handling for the arg class or lifting this [SO answer](https://stackoverflow.com/a/35483750) could resolve the serialization errors in  the `_serialise` classmethod  
 https://github.com/cunnane/xloil/blob/2b99fc7f091a784a012a69762320c57a8bd5f60c/libs/xlOil_Python/Package/xloil/jupyter_kernel.py#L22 

On the Excel side I think that ensuring args can handle a lack of type decoration in `_FuncDescriptionDecoder` should be sufficient.
Currently without a `typeof `present the deserialization in `object_hook` doesn't cast to an Arg Object and just returns a dict which breaks `arg_to_funcarg` 

https://github.com/cunnane/xloil/blob/2b99fc7f091a784a012a69762320c57a8bd5f60c/libs/xlOil_Python/Package/xloil/jupyter.py#L28C1-L35C19 

```
    arg_type = arg.typeof
               ^^^^^^^^^^
AttributeError: 'dict' object has no attribute 'typeof'
```

Thanks again for making this all available!

**Comment by cunnane** [comment]
Thanks for reporting this in detail!  It should be fixed in 0.18.0 which I've just released

**Comment by andyvandy** [comment]
Much appreciated!!
